api/src/services/versions.ts

```diff
@@ -29,21 +28,28 @@ export class VersionsService extends ItemsService {
 	}
 
 	private async validateCreateData(data: Partial<Item>): Promise<void> {
-		if (!data['key']) throw new InvalidPayloadError({ reason: `"key" is required` });
+		const versionCreateSchema = Joi.object({
+			key: Joi.string().required(),
+			name: Joi.string().allow(null),
+			collection: Joi.string().required(),
+			item: Joi.string().required(),
+		});
+
+		const { error } = versionCreateSchema.validate(data);
+		if (error) throw new InvalidPayloadError({ reason: error.message });
 
 		// Reserves the "main" version key for the version query parameter
 		if (data['key'] === 'main') throw new InvalidPayloadError({ reason: `"main" is a reserved version key` });
 
-		if (!data['collection']) {
-			throw new InvalidPayloadError({ reason: `"collection" is required` });
+		try {
+			await this.authorizationService.checkAccess('read', data['collection'], data['item']);
+		} catch {
+			throw new ForbiddenError();
 		}
```

Comment:
I believe this `try/catch` is unnecessary as `checkAccess` throws a `ForbiddenError` which we are re-throwing anyways.

```suggestion
			await this.authorizationService.checkAccess('read', data['collection'], data['item']);
```

Reply:
This function is dropped in v11, so this will need to be refactored.

Reply:
In that case we should refactor to use `validateAccess` which is in v11.

Reply:
Unfortunately, can't remember exactly (should have added a comment) but I think it is necessary to catch & hide the error when the given collection itself doesn't exist. Have you tried that?

Reply:
Checking this case it throws an `An unexpected error occurred` instead of the expected `You don't have permission to access this.` so this is indeed needed!